07-space.Rmd

```diff
@@ -117,13 +120,26 @@ plot(zjd)
 
 ## CRSs
 
+A CRS is a coordinate reference system, used for plotting data on maps. There are many systems in use but they can generally be classified into two groups; 'projected' and 'geographic'. A projected system, such as Eastings/Northings, plots locations onto a flat 2D projection of the Earth's surface. A geographic system, such and Longitude/Latitude, refers to locations on the 3D surface of the globe. Distance and direction calculations work differently between geographc and projected CRSs, so it it often necessary to convert from one to another. Fortunately R makes this very easy, and every CRS has its own unique refernce number, for example 27700 for the British National Grid system. 
+
 Get and set Coordinate Reference Systems (CRSs) with the command `st_crs()`.
 Transform CRSs with the command `st_transform()`, as demonstrated in the code chunk below, which converts the 'lon/lat' geographic CRS of `crashes_sf` into the projected CRS of the British National Grid:
 
 ```{r crs1}
 crashes_osgb = st_transform(crashes_sf, 27700)
 ```
 
+When you define the gemotery in a data frame, it creates a multipoint vector. This is a single column containing coordinate pairs for point data, or multiple coordinate pairs for line vecotrs or polygons. Sometimes it may be necessary to convert from one CRS to another without creating a multipoint vector. The following code is one way to convert Eastings/Northings (CRS 27700) to Latitude/Longitude (CRS 4326) without creating a multipoint vector.
+
+```{r}
+#data frame name is Crashes, containing location data in separate columns for Eastings and Northings
+latlongcoords = (cbind(Crashes$Easting , Crashes$Northing))
+latlong = SpatialPointsDataFrame(latlongcoords,data = Crashes ,proj4string = CRS("+init=epsg:27700"))
```

Comment:
Can you use the `sf` equivalent:

```
latlong = sf::st_as_sf(Crashes, coords = c("Easting", "Northing"), crs = 27700)
```

???

Reply:
When I tried this in the script I got an error after the next line:
unable to find an inherited method for function ‘spTransform’ for signature ‘"sf", "CRS"’